### PR TITLE
[Ftr] master feature

### PR DESCRIPTION
**Pull request by cityiron**
**What this PR does**:
Master feature, support one api config accept almost all of  request to immutable target cluster.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by zhangshen023 — APPROVED** [review]

**Review by williamfeng323 — APPROVED** [review]

File: pkg/client/client.go [hunk 1]
```diff
@@ -26,3 +26,133 @@ type Client interface {
...
+func (opt *CommonOption) Usable() bool {
+	return opt.usable
+}
+
+// SetUsable set usable
+func (opt *CommonOption) SetUsable(b bool) {
+	opt.usable = b
+}
+
+type dubboParamTypesOpt struct {
```
Comment (williamfeng323) [inline pkg/client/client.go:68]: should we place these client-specific options back to the dubbo package?

File: pkg/config/api_config.go [hunk 2]
```diff
@@ -180,6 +180,14 @@ type IntegrationRequest struct {
 type MappingParam struct {
 	Name  string `json:"name" yaml:"name"`
 	MapTo string `json:"mapTo" yaml:"mapTo"`
+	Opt   Opt    `json:"opt,omitempty" yaml:"opt,omitempty"`
+}
+
+// Opt option, action for compatibility.
+type Opt struct {
+	Name   string `json:"name" yaml:"name"`
+	Open   bool   `json:"open" yaml:"open"`
```
Comment (williamfeng323) [inline pkg/config/api_config.go:189]: we might need some descriptions on the field Open and Usable

File: pkg/client/dubbo/dubbo.go [hunk 3]
```diff
@@ -154,17 +154,32 @@ func (dc *Client) Call(req *client.Request) (res interface{}, err error) {
...
+}
+
 // MapParams params mapping to api.
 func (dc *Client) MapParams(req *client.Request) (interface{}, error) {
 	r := req.API.Method.IntegrationRequest
 	var values []interface{}
 	for _, mappingParam := range r.MappingParams {
+		var opt client.IOption
+		if mappingParam.Opt.Open {
+			opt = client.DefaultMapOption[mappingParam.Opt.Name]
```
Comment (williamfeng323) [inline pkg/client/dubbo/dubbo.go:174]: add validation to ensure the option [mappingParam.Opt.Name] is supported?

File: pkg/client/dubbo/dubbo.go [hunk 4]
```diff
@@ -154,17 +154,32 @@ func (dc *Client) Call(req *client.Request) (res interface{}, err error) {
...
+	return types, values, nil
+}
+
 // MapParams params mapping to api.
 func (dc *Client) MapParams(req *client.Request) (interface{}, error) {
 	r := req.API.Method.IntegrationRequest
 	var values []interface{}
 	for _, mappingParam := range r.MappingParams {
+		var opt client.IOption
+		if mappingParam.Opt.Open {
```
Comment (williamfeng323) [inline pkg/client/dubbo/dubbo.go:173]: add unit-tests for this 'if'?

File: pkg/client/client.go (hunk 1 above)
Comment (cityiron) [reply pkg/client/client.go:68]: I think so. But I not sure http is need this.

**Review by yqxu — APPROVED** [review]

**Pull request merged**